dask_expr/tests/test_merge.py

```diff
@@ -1083,3 +1083,17 @@ def test_merge_tuple_left_on():
         df.merge(df, on=("a",)),
         check_index=False,
     )
+
+
+def test_merged_partitions_filtered():
+    a = from_dict({"x": range(1000), "y": [1, 2, 3, 4] * 250}, npartitions=10)
+    b = from_dict({"xx": range(100), "yy": [1, 2] * 50}, npartitions=3)
+
+    result = a.partitions[:5].merge(
+        b, left_on=["y"], right_on=["yy"], how="inner", broadcast=True
+    )
+    expr = result.optimize(fuse=False).expr
+    assert not expr._filtered
+    assert expr.left._filtered
+    assert expr.divisions == expr._divisions()
+    assert len(expr.divisions) == 6
```

Comment:
can you check the result as well?

Reply:
Good call.

While adding this check I actually noticed that divisions are often "wrong" after a merge (unrelated to the changes in this PR). More specifically, it seems like we are inheriting divisions after a broadcast join, even if we aren't merging on the index. If I understand correctly, we inherit the partition count in this case, but the divisions are likely to change.

Reply:
Yeah that's correct, we probably should only inherit on index merges, pandas is not consistent with the index when you merge on columns :(